File: shoring/compat/src/main/java/com/panopset/compat/HiddenFolder.java
package com.panopset.compat;

public enum HiddenFolder {

  INSTANCE;

  public static String getFullPathRelativeTo(String relativePathToAppend) {
    return INSTANCE.getBasePath() + Stringop.FILE_SEP +  relativePathToAppend;
  }

  private String getBasePath() {
    return Stringop.USER_HOME + Stringop.FILE_SEP + getHiddenFolderName();
  }

  private String rootLogFileName;

  public static String getRootLogFileName() {
    return INSTANCE.getRlfn();
  }

  private String getRlfn() {
    if (rootLogFileName == null) {
      rootLogFileName = "root";
    }
    return rootLogFileName;
  }

  public static void setRootLogFileName(String value) {
    INSTANCE.rootLogFileName = value;
  }

  private String hiddenFolderName;

  public static String getHiddenFolderName() {
    return INSTANCE.getHfn();
  }

  private String getHfn() {
    if (hiddenFolderName == null) {
      hiddenFolderName = "temp_but_should_replace";
    }
    return hiddenFolderName;
  }

  public static void setHiddenFolderName(String value) {
    INSTANCE.hiddenFolderName = value;
  }
}

File: code/shoring/compat/src/main/kotlin/com/panopset/compat/HiddenFolder.kt
package com.panopset.compat

import com.panopset.compat.Stringop.FSP
import com.panopset.compat.Stringop.USH

object HiddenFolder {

    var companyName = "should_have_been_replaced_by_BrandedApp"
    var rootLogFileName = "root"

    fun getFullPathRelativeTo(relativePathToAppend: String): String {
        return "$USH${FSP}Documents$FSP$companyName$FSP$relativePathToAppend"
    }
}
